src/cpp/src/image_generation/stable_diffusion_xl_pipeline.hpp

```diff
@@ -443,11 +451,25 @@ class StableDiffusionXLPipeline : public DiffusionPipeline {
             // check whether scheduler returns "denoised" image, which should be passed to VAE decoder
             const auto it = scheduler_step_result.find("denoised");
             denoised = it != scheduler_step_result.end() ? it->second : latent;
+
+            if (do_callback) {
+                if (callback(inference_step, denoised)) {
+                    ov::Shape output_shape = {1,
+                                              generation_config.height / vae_scale_factor,
+                                              generation_config.width/ vae_scale_factor,
+                                              3};
+                    return ov::Tensor(ov::element::u8, output_shape);
```

Comment:
should not we return empty tensor? empty means with empty shape `{}` to denote "invalid" tensor

Reply:
> Segmentation fault (core dumped)

Can we fix it to provide meaningful error? E.g. ensure that `imwrite` does not fail on empty image and call imwrite only if image.get_size() > 0